radioapp/ui/ui_manager.py

```diff
@@ -1,45 +1,47 @@
-import tkinter
-from tkinter import ttk
 from .home import UIHome
-from .settings import UISettings
+import tkinter as tk
+
+
+class UIManager:
+    """Manages UI Screens and Settings."""
 
-class UI():
-    """Holds the main window."""
-    
     def __init__(self):
-         """
-         Initializes window and frame.
-         
-         Set defaults for the tkinter instance.
-         """
-        
-        self.window = tkinter.Tk()
-        self.window.columnconfigure(0, weight=1, uniform=1)
-        self.window.rowconfigure(0, weight=1, uniform=1)
+        """Creates a window, and manages different screens and settings."""
+
+        self.current_slider_number = 0
+        self.current_screen = None
+
+        self.initialize_window()
+
+        self.switch_screen(self.display_home, 0)
+
+    def display_home(self):
+        """Displays the Home screen."""
+
+        self.current_screen = UIHome(self.window, self)
+        self.current_screen.pack_screen()
+
+    def initialize_window(self):
+        """Initializes the main window."""
+
+        self.window = tk.Tk()
         self.window.geometry("160x128")
-        self.window.resizable(0,0)
+        self.window.configure(background="#3399ff")
+        self.window.resizable(0, 0)
         self.window.pack_propagate(0)
-        
-        #Set defaults for the tkinter instance.
-        self.style = ttk.Style(self.window)
-        self.style.configure('TFrame', background='blue')
-        self.window.option_add("*Background", "blue")
-        self.window.option_add("*Foreground", "white")
-
-        self.start()
-        
-    def start(self):
-        self.go_home()
-        self.window.mainloop()
-
-    def go_home(self):
-        self.currentPage = UIHome(self.window, self.go_settings, None, None)
 
-    def go_settings(self):
-        self.currentPage = UISettings(self.window, self.go_home)
+    def switch_screen(self, screen_class, action_number):
+        """Switches between Screen."""
 
+        if action_number == 1:
```

Comment:
I'm curious what `action_number` is. Could you document it in the docstring?

Reply:
Action number is passed down from various UI screens that tells UI Manager to do perform certain action. In this case, the action number 1 is passed down form U Backlight Screen that tells the UI Manager to save the slider number from UI Backlight Screen to UI Manager.

Reply:
We may just want to have a callback function instead of the action number, that way we don't have the manager holding code that isn't managing UI. Or at least an Enum instead of a #. Message me on slack if you need help with this.

Reply:
The manager is not holding any code. Actions are performed only when we are switching screens. Sometimes actions are not performed at all. Actions depend on what the current screen tells the UI Manager to do. In the case of Backlight screen, we need to perform the action where slider number is saved. The action number is passed down from the current screen to UI Manager. I don't see anything wrong with action number. Everything else really makes it complicated, and there is no reason to do that. It'll become clearer when the Backlight is added.